prime-router/docs/proposals/lineage/README.md

```diff
@@ -0,0 +1,123 @@
+
+A **Report** in the hub is simply a set of **Item**, with additional metadata.   For example, if the data is in CSV format, the Hub expects a single row of Column Header strings, followed by one or more rows of **Item** data.    It is acceptable for a Report to have just a single Item.  A Report with no Items is not processed.
+
+The Hub implements a simple workflow engine.   Each step in the engine is a **Task**.  Most (but not all) tasks create one or more Reports. Current **Task** in the system include: 
+
+| Task      |     Description
+| ---       | ---
+| Receive   | Receive a new Report into the Hub
+| Transform | Validate a received report, and transform into desired output formats, according to configured data filter business rules
+| Batch     |  Merge one or more reports together.   Typically done to consolidate data per receiver requirements.
+| Send      | Reliably delivery data to an external receiver, eg, an API, sftp site, or provide for upload (and notify with an email)
+| Wipe      | Remove data, per business rules
+
+An **Organization** can recieve and/or send data to the hub.  A typical Organization is a state or local Public Health Department.   Strac, Redox, and SimpleReport are all also Organizations in the hub.
+
+An **OrganizationService** is an endpoint to which the Hub can send a specific kind of data to a specific place, so that the Organiation can use the data.  For example, an OrganizationService might be an sftp site where a local health department wants hl7 files uploaded every 4 hours.
+
+Likewise, an **OrganizationClient** is an endpoint that sends data to the Hub API.   For example, SimpleReport has an OrganizationClient that can send .csv data to the hub, as often as they like.
+
+
+## Assumptions
+
+- TASKs create zero or more REPORTs which have one or more ITEMs
+- ITEMs are like quarks.They only exist inside of a REPORT.  Its fine and quite common to have a report with one item.
+- We must track lineage per-item, however, people think in terms of REPORTs.  Hence we need to track both.
+- Tracking of sending- and receiving- Organizations is crucial, to know who gets permissions to see the data.
+- Many:many -- A given ACTION could take in many reports, and produce many reports.  Need to allow for both splitting and merging.
+- Many:many -- An ITEM can be the result of multiple parent items, and can result in multiple child ITEMs.
+- A few schema best practices:  Every table must have a single col PK (either a serial or a UUID), and a created_date, Normalize independent data.   Use singular nouns.
+-- Receipt of a dataset implies the right to see the lineage of that dataset.
+-- Sending a dataset implies the right to see the lineage of that dataset. 
+
+## Proposed Design
+
+Please [pull up the ER diagram](./lineage-er-diagram.jpeg) for the detailed proposed schema.
+
```

Comment:
I've been using this SQL style guide https://about.gitlab.com/handbook/business-ops/data-team/platform/sql-style-guide. Read it over, if you agree with its recommendations, then we can check it in. One thing I noticed in your schema was the use of `_date` vs ` _at` for timestamps.

Reply:
Switched to `_at`.